src/features/post/detail/DetailPage.tsx

```diff
@@ -1,16 +1,36 @@
 import React from 'react';
-import { Box, Button, Avatar, Container, Typography } from '@mui/material';
+import { Box, Button, Avatar, Container, Typography, Menu, MenuItem, Fade, IconButton } from '@mui/material';
 import { orange, grey } from '@mui/material/colors';
 import { ThumbUpAlt, RemoveRedEye, Comment } from '@mui/icons-material';
 import { GetDetailResponse, postApi } from '@/api';
 import { GetServerSideProps } from 'next';
 import ReplyList from '@/components/reply/ReplyList';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import { useRouter } from 'next/router';
+
 
 interface DetailProps {
   detail: GetDetailResponse;
 }
 
 const DetailPage = ({ detail }: DetailProps) => {
+  const router = useRouter();
+  const id = Number(router.query?.id);
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const openDetailMenu = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const onClickEdit = () => {
+    router.push(`/post/edit/${id}`)
+  };
+  const onClickDelete = () => {
+  };
```

Comment:
eslint 설정으로 빈 함수는 오류가 나게 설정되어 있어서 빌드할 때는 아마 오류가 날 것 같아요.
아래처럼 일단 해주셔도 될 것 같고 삭제 api 연동은 추후에 제가 작업을 할게요!
```ts
// eslint-disable-next-line @typescript-eslint/no-empty-function
  const onClickDelete = () => {};
```

Reply:
가독성을 위해 함수 사이에 공백을 추가해도 좋을거 같아요
```suggestion
  const openDetailMenu = Boolean(anchorEl);
  
  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };
  
  const handleClose = () => {
    setAnchorEl(null);
  };
  
  const onClickEdit = () => {
    router.push(`/post/edit/${id}`)
  };
  
  const onClickDelete = () => {
  };
```

Reply:
아하! 넵 알겠습니다!!